surveys/2021-annual-survey/questions.md

```diff
@@ -649,6 +649,33 @@ Select one:
 > This question assess hiring sentiment. Although there is intrinsic uncertainty, it is easy to answer and forward looking.
 > It will also be interesting to see what the demand for Rust skills from companies is over time.
 
+### Rate how much the follow statements are reasons your team use Rust at work.
+
+Statements:
+
+- We use Rust for its performance characteristics.
+- We use Rust because we need to run software in a constrained or specialized environment.
```

Comment:
This feels a bit underspecified - do you mean resource-constrained? "Specialised" is pretty generic too

Reply:
Resource constrained is what I had in mind, but I don't want to be too specific. I'm trying to get at the sentiment of "We need specific and direct control over how the end binary behaves, and Rust gives us that control". This differs from performance in that it's not about making it faster, consume less memory, etc., it's about making sure you have predictable and controllable behavior.

Reply:
Hmm, I feel that "predictable and controllable" is orthogonal to the environment in which the program is run. Whether you consider that separate from performance is an interesting question - IMO they are both about performance, one is minimizing resource usage and one is having predictable resource usage. But they are distinct enough to have separate answers. I wonder if we also need an answer specific to the environment, e.g., things like setting memory-mapped registers is impossible in a lot of languages and isn't about performance or resource usage.